stackslib/src/net/api/getclaritymetadata.rs

```diff
@@ -0,0 +1,217 @@
+    StacksHttpRequest, StacksHttpResponse,
+};
+use crate::net::{Error as NetError, StacksNodeState, TipRequest};
+
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+pub struct ClarityMetadataResponse {
+    pub data: String,
+}
+
+#[derive(Clone)]
+pub struct RPCGetClarityMetadataRequestHandler {
+    pub clarity_metadata_key: Option<String>,
+    pub contract_identifier: Option<QualifiedContractIdentifier>,
+}
+impl RPCGetClarityMetadataRequestHandler {
+    pub fn new() -> Self {
+        Self {
+            clarity_metadata_key: None,
+            contract_identifier: None,
+        }
+    }
+}
+
+/// Decode the HTTP request
+impl HttpRequest for RPCGetClarityMetadataRequestHandler {
+    fn verb(&self) -> &'static str {
+        "GET"
+    }
+
+    fn path_regex(&self) -> Regex {
+        Regex::new(&format!(
+            r"^/v2/clarity_metadata/(?P<address>{})/(?P<contract>{})/(?P<clarity_metadata_key>(analysis)|({}))$",
+            *STANDARD_PRINCIPAL_REGEX_STRING,
+            *CONTRACT_NAME_REGEX_STRING,
+            *METADATA_KEY_REGEX_STRING
+        ))
+        .unwrap()
+    }
+
+    fn metrics_identifier(&self) -> &str {
+        "/v2/clarity_metadata/:principal/:contract_name/:clarity_metadata_key"
+    }
+
+    /// Try to decode this request.
+    /// There's nothing to load here, so just make sure the request is well-formed.
+    fn try_parse_request(
+        &mut self,
+        preamble: &HttpRequestPreamble,
+        captures: &Captures,
+        query: Option<&str>,
+        _body: &[u8],
+    ) -> Result<HttpRequestContents, Error> {
+        if preamble.get_content_length() != 0 {
+            return Err(Error::DecodeError(
+                "Invalid Http request: expected 0-length body".to_string(),
+            ));
+        }
+
+        let contract_identifier = request::get_contract_address(captures, "address", "contract")?;
+        let metadata_key = request::get_clarity_key(captures, "clarity_metadata_key")?;
+
+        self.contract_identifier = Some(contract_identifier);
+        self.clarity_metadata_key = Some(metadata_key);
+
+        let contents = HttpRequestContents::new().query_string(query);
+        Ok(contents)
+    }
+}
+
+/// Handle the HTTP request
+impl RPCRequestHandler for RPCGetClarityMetadataRequestHandler {
+    /// Reset internal state
+    fn restart(&mut self) {
+        self.contract_identifier = None;
+        self.clarity_metadata_key = None;
+    }
+
+    /// Make the response
+    fn try_handle_request(
+        &mut self,
+        preamble: HttpRequestPreamble,
+        contents: HttpRequestContents,
+        node: &mut StacksNodeState,
+    ) -> Result<(HttpResponsePreamble, HttpResponseContents), NetError> {
+        let contract_identifier = self.contract_identifier.take().ok_or(NetError::SendError(
+            "`contract_identifier` not set".to_string(),
+        ))?;
+        let clarity_metadata_key = self.clarity_metadata_key.take().ok_or(NetError::SendError(
```

Comment:
Clarity metadata keys have a well-defined structure, so the RPC endpoint should return HTTP 400 if the metadata key is ill-formed.  You will need to expand this method (or `try_parse_request` -- up to you) to validate the structure of the Clarity metadata key, including determining whether or not the key's `StoreType` and `var_name` values are supported values.  `var_name` will, unfortunately, take some effort because the Clarity DB codebase uses bare string literals in its calls to `ClarityDatabase::make_metadata_key()` instead of enums, so your PR should address this as well.

Reply:
Apparently the metadata `var_name` can be an arbitrary string, where here it's a [variable_name](https://github.com/stacks-network/stacks-core/blob/26c448727def84bbef34ba4d14ec4fd5244a2390/clarity/src/vm/database/clarity_db.rs#L1329), a [map_name](https://github.com/stacks-network/stacks-core/blob/26c448727def84bbef34ba4d14ec4fd5244a2390/clarity/src/vm/database/clarity_db.rs#L1467), a [token_name](https://github.com/stacks-network/stacks-core/blob/26c448727def84bbef34ba4d14ec4fd5244a2390/clarity/src/vm/database/clarity_db.rs#L1811)).

Can we really use an enum here?

Reply:
Yes; however, some reserved `var_name` strings can only be paired with certain `StoreType`s.  For example, the use of `StoreType::Contract` would require `var_name` to be `contract-size`, `contract-src`, `contract-data-size`, or `contract`, but nothing else.

One way to implement these constraints could be to implement an enum to capture all of the valid `StoreType` / `var-name` pairings, and permit only the `StoreType` variants which allow arbitrary `var_name` values to have arbitrary `var_name` values.